ark-core/src/proof_of_funds.rs

```diff
@@ -46,6 +49,8 @@ pub struct Input {
     pk: XOnlyPublicKey,
     spend_info: (ScriptBuf, taproot::ControlBlock),
     is_onchain: bool,
+    // FIXME: make the input type an enum based on the input type.
+    preimage: Option<[u8; PREIMAGE_LENGTH]>,
```

Comment:
👍 Yep, I think we need to get rid of these two fields and model this as an enum.